retriever.py

```diff
@@ -13,14 +16,69 @@
 from preproc import get_functions
 from tokenizer import Tokenizer
 from transformers import AutoModelForCausalLM
+from gen_pr_items import PR_ITEMS
 
+REPO_ROOT = Path(__file__).resolve().parent
 
-class Retriever:
-    def __init__(self, experiment_name):
-        self.experiment_name = experiment_name
+
+class EmbeddingGenerator:
+    """
+    Generate embedding for an item based using the model
+    """
+
+    def __init__(self):
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
         self.config = TDArgs()
-        assets_path = os.path.join("assets", self.experiment_name)
 
+        generator = Llama.build(
+            ckpt_dir=os.path.expanduser(self.config.model_ckpt_dir),
+            tokenizer_path=os.path.expanduser(self.config.tokenizer_path),
+            max_seq_len=self.config.max_context_len,
+            max_batch_size=self.config.max_batch_size,
+            use_kv_cache=False,
+            model_parallel_size=1,
+        )
+        self.model = generator.model.to(self.device)
+        self.tokenizer = Tokenizer(self.config)
+
+    def gen_embedding(self, items: List[str]) -> List[Any]:
```

Comment:
Why did we move this logic out of the retriever class? I feel like this should be a function inside retriever

Reply:
I have one way to parse a PR but multiple indexes that I want to test it against, so I wanted to separate the generation of the embedding for the parsed PR from the evaluation of this PR embedding with the index